assets/js/ledyer-checkout-for-woocommerce.js

```diff
@@ -444,6 +444,14 @@ jQuery(function ($) {
             $(document).on('ledyerCheckoutOrderComplete', function (event) {
                 lco_wc.logToFile('ledyerCheckoutOrderComplete from Ledyer triggered');
 
+                if (lco_wc.redirectUrl !== null) {
+                    // This means that placeLedyerOrder was called successfully already
+                    // (Due to an earlier call caused by client validation)
+                    lco_wc.redirectUrl.searchParams.append('lco_purchase_complete', 'yes');
+                    window.location.href = lco_wc.redirectUrl.toString();
+                    return;
+                }
+
```

Comment:
If client validation was triggered, we call `placeLedyerOrder`, which creates and validates the woo order.

Later on, after validation has been sent to Ledyer, Ledyer Checkout will trigger the event `ledyerCheckoutOrderComplete` to give the final OK. 

This function previously would always trigger `placeLedyerOrder`, since we initially didn't have the validation step and that was the only place to call placeLedyerOrder.

If validation was triggered however, we don't wanna run `placeLedyerOrder` function twice (as that will create a duplicate order in Woo), so we'll save the redirect url that was generated there in the first execution and when the order complete event is received here, then redirect immediately because the order exists in  Woo.

Reply:
Maybe we don't actually need to append
`lco_wc.redirectUrl.searchParams.append('lco_purchase_complete', 'yes');`

As `lco_confirm` is already appended.

It was something I experimented with to make sure we run the confirmation code at the right time.

It is possible it works with just lco_confirm, need to do an extra round of testing.